src/pyosmeta/github_api.py

```diff
@@ -0,0 +1,216 @@
+"""
+A module that houses all of the methods related to interfacing
+with the GitHub API. THere are three groupings of activities here:
+
+1. Parsing github issues to return peer review information
+2. Parsing contributor profile data to return names and affiliations where
+available
+3. Parsing package repositories to return package metadata such as pull request
+numbers, stars and more "health & stability" related metrics
+"""
+
+import logging
+import os
+
+import requests
+from dataclasses import dataclass
+from dotenv import load_dotenv
+from typing import Any
+
+
+@dataclass
+class GitHubAPI:
```

Comment:
I wonder if this makes more sense to name GitHubMetadata? Anything dealing with metadata from GitHub could then go into this file. Maybe renaming the file to `github_metadata.py`.

PS @lwasser I love your approach and how you are thinking about how the next developer will react to the code.

Reply:
Right now i'm only moving methods into this module that interface with the github API. and then all of the actual metadata parsing is happening in the issues and contributors modules. so i'm a bit hesitant on naming it metadata here as well. this module is for authentication and get requests to the API only (as i've moved things anyway now). this will be the hardest module to write tests for as well. 😆